From 727d3cea49caf5f8c47e00dd0c6a8f7de53198f1 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Tue, 17 Sep 2019 03:24:11 -0700
Subject: [PATCH] _webos_init_install_vars: add variables originally imported
 from pro

:Release Notes:
Add webos_customizationdir, webos_developerdir, webos_emulatorshareddir
webos_firstusesentinelfile, webos_settingsservice_errorsentinelfile
instead of patching it in through the recipe.

:Detailed Notes:
Finish the migration from pro as mentioned in:
https://gpro.lgsvl.com/190824

:Testing Performed:
Only build tested.

:QA Notes:

:Issues Addressed:
[PLAT-85921] Move /opt into /usr/opt

Change-Id: Ib5233ffff417a3920644ad5bcdc5d5cb542c4203
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/cmake-modules-webos/+/255718 _webos_init_install_vars: add variables originally imported from pro]

 webOS/webOS.cmake | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/webOS/webOS.cmake b/webOS/webOS.cmake
index 7f64122..e72e7df 100644
--- a/webOS/webOS.cmake
+++ b/webOS/webOS.cmake
@@ -490,6 +490,23 @@ macro(_webos_init_install_vars)
 	# The /opt/webos/tests tree contains all installed unit and/or integration test scripts and executables
 	_webos_set_from_env(WEBOS_INSTALL_TESTSDIR                  webos_testsdir                  ${WEBOS_INSTALL_ROOT}/opt/webos/tests)
 
+	# This tree contains subdirectories of various types of customization data
+	_webos_set_from_env(WEBOS_INSTALL_WEBOS_CUSTOMIZATIONDIR              webos_customizationdir                  ${WEBOS_INSTALL_PREFIX}/palm/customization)
+
+	# This is the root of the tree that is accessible to developers who log into a
+	# device with ssh when it is in developer mode. It is where their side-loaded
+	# (webOS and native) apps will be installed for debugging. Everything
+	# under this tree is erased by a NYX_SYSTEM_ERASE_DEVELOPER erasure.
+	_webos_set_from_env(WEBOS_INSTALL_DEVELOPERDIR                        webos_developerdir                      ${WEBOS_INSTALL_MEDIADIR}/developer)
+
+	# This directory is shared by the emulator for network mounting by its host OS
+	_webos_set_from_env(WEBOS_INSTALL_EMULATORSHAREDDIR                   webos_emulatorshareddir                 ${WEBOS_INSTALL_MEDIADIR}/shared)
+
+	# The presence of this file indicates that First Use has been completed.
+	_webos_set_from_env(WEBOS_INSTALL_FIRSTUSESENTINELFILE                webos_firstusesentinelfile              ${WEBOS_INSTALL_SYSMGR_LOCALSTATEDIR}/preferences/ran-firstuse)
+
+	# Path to file which indicate failure to init settingsservice
+	_webos_set_from_env(WEBOS_INSTALL_SETTINGSSERVICE_ERRORSENTINELFILE   webos_settingsservice_errorsentinelfile ${WEBOS_INSTALL_WEBOS_LOCALSTATEDIR}/settingsservice_critical_error)
 endmacro()
 
 
